File: src/main/java/com/project/moodanalyzer/entity/Song.java
package com.project.moodanalyzer.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "songs")
public class Song {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private String title;
    private String artist;

    private Double bpm;
    private Double pitch;
    private Double energy;

    private String mood;

    private String filePath;
}

File: backend/models/Song.js
import mongoose from "mongoose";

const songSchema = new mongoose.Schema({
  title: { type: String, required: true },
  artist: { type: String },
  mood: { type: String },
  filePath: { type: String, required: true },
});

const Song = mongoose.model("Song", songSchema);
export default Song;
